app/components/price-graph/price-graph.tsx

```diff
@@ -48,46 +48,39 @@ export const PriceGraphDataInjected = () => {
   const [graphRange, setGraphRange] = React.useState<GraphRangeType>(GraphRange.ONE_DAY)
 
   const { error, loading, data, refetch } = useBtcPriceListQuery({
-    fetchPolicy: "no-cache",
+    fetchPolicy: "network-only",
+    variables: { range: graphRange },
   })
+  const priceList = data?.btcPriceList ?? []
 
-  if (loading || data === null || !data?.btcPriceList) {
+  React.useEffect(() => {
+    refetch({ range: graphRange })
+  }, [graphRange, refetch])
+
+  if (loading || data === null || data?.btcPriceList === null) {
     return <ActivityIndicator animating size="large" color={palette.lightBlue} />
   }
 
   if (error) {
     return <Text>{`${error}`}</Text>
   }
 
-  const lastPrice = data.btcPriceList[data.btcPriceList.length - 1]
+  const ranges = GraphRange[graphRange]
+  const timestamps = [300, 1800, 86400, 86400, 86400]
+
+  const lastPrice = priceList[priceList.length - 1]
   if (!loading && lastPrice) {
     const unixTime = Date.now() / 1000
-    if (graphRange === GraphRange.ONE_DAY) {
-      if (unixTime - lastPrice.timestamp > 300) {
-        refetch({ range: GraphRange.ONE_DAY })
-      }
-    } else if (graphRange === GraphRange.ONE_WEEK) {
-      if (unixTime - lastPrice.timestamp > 1800) {
-        refetch({ range: GraphRange.ONE_WEEK })
-      }
-    } else if (graphRange === GraphRange.ONE_MONTH) {
-      if (unixTime - lastPrice.timestamp > 86400) {
-        refetch({ range: GraphRange.ONE_MONTH })
-      }
-    } else if (graphRange === GraphRange.ONE_YEAR) {
-      if (unixTime - lastPrice.timestamp > 86400) {
-        refetch({ range: GraphRange.ONE_YEAR })
-      }
-    } else if (graphRange === GraphRange.FIVE_YEARS) {
-      if (unixTime - lastPrice.timestamp > 86400) {
-        refetch({ range: GraphRange.FIVE_YEARS })
-      }
+    const index = ranges.indexOf(graphRange)
+    if (unixTime - lastPrice.timestamp > timestamps[index]) {
+      setGraphRange(ranges)
+      refetch({ range: ranges })
```

Comment:
are we refetching twice right now? here and https://github.com/GaloyMoney/galoy-mobile/pull/979/files#diff-d75c8934e23c87b556deda51f102e3d3045a1da88266c8a2391ba51c8a130873R57 ?

Reply:
Oops! You're right. The useEffect is unnecessary.